R Syntax: split analysis and analysisForm

boutinb opened the pull request:
As described in the JASP Architecture for R Syntax document, we need to split the analysis and analysisForm objects. For this the AnalysisFormBase is added (AnalysisBase existed already), the Base objects contain all calls from Analysis to AnalysisForm and vice-versa.

The initColumnWithString is function is moved to the DataSet object so that the RDataFrame importer can use it.

The parseOptions function in the AnalysisForm is already added.

JorisGoosen commented on QMLComponents/analysisform.cpp:
```diff
@@ -272,16 +273,28 @@ void AnalysisForm::setHasVolatileNotes(bool hasVolatileNotes)
-	return false;
+	jsonResult["options"] = jsonOptions;
+	jsonResult["error"] = fq(getError());
+	return tq(jsonResult.toStyledString());
```
Why not return the Json value then?

boutinb replied:
In the jaspQMLR example, the QML form is just a QuickItem object, so I call the parseOptions with the QMetaObject::invokeMethod method:
https://github.com/jasp-stats/jaspQmlR/blob/a7a93a2f6fa103d47560c1ed541e9c252eed3671/src/qmlR.cpp#L208
In this way I don't need to load the AnalysisFormBase header in the jaspQMLR package.
I don't think I can pass a Json::Value argument in the InvokeMethod, so I thought just send the string, and reset it as Json object afterwards.

boutinb committed:
Prepare R Syntac

Move the initColumnWithStrings from DataSetPackage to DataSet.
Add parseOptions in AnalysisForm
Update generateWrapper so that it can be called from the

JorisGoosen commented on QMLComponents/analysisform.cpp:
```diff
@@ -272,16 +270,28 @@ void AnalysisForm::setHasVolatileNotes(bool hasVolatileNotes)
+	Json::Value	 jsonOptions;
+	Json::Value jsonResult(Json::objectValue);
+
+	QJsonDocument doc = QJsonDocument::fromJson(options.toUtf8());
```
Why is this step here?
It seems that the whole QJsonDocument

JorisGoosen committed:
remove useless QJsonDocument

JorisGoosen commented on QMLComponents/analysisform.cpp:
```diff
@@ -135,11 +133,10 @@ void AnalysisForm::runScriptRequestDone(const QString& result, const QString& co
 				// For this, bind all controls to null and wait for the controls to be completely destroyed.
 				QTimer::singleShot(0, [=](){
-					bindTo(options);
+				bindTo(options);
```
```suggestion
					bindTo(options);
```

JorisGoosen commented on QMLComponents/analysisform.cpp:
```diff
@@ -135,11 +133,10 @@ void AnalysisForm::runScriptRequestDone(const QString& result, const QString& co
+				bindTo(options);
 					blockValueChangeSignal(false, false);
-					_analysis->boundValueChangedHandler();
+				_analysis->boundValueChangedHandler();
```
```suggestion
					_analysis->boundValueChangedHandler();
```

JorisGoosen committed:
Tabs

JorisGoosen commented on QMLComponents/analysisform.cpp:
```diff
@@ -567,7 +575,7 @@ void AnalysisForm::setTitle(QString title)
 }
 
-void AnalysisForm::setOptionNameConversion(const QVariantList & conv)
+void AnalysisForm::setOptionNameConversion(const QVariantList &conv)
```
```suggestion
void AnalysisForm::setOptionNameConversion(const QVariantList & conv)
```

JorisGoosen committed:
spacen

JorisGoosen commented on QMLComponents/analysisform.h:
```diff
@@ -145,6 +145,7 @@ public slots:
 	Q_INVOKABLE void		refreshAnalysis();
 	Q_INVOKABLE bool		initialized()			const	{ return _initialized; }
 	Q_INVOKABLE QString		generateWrapper()		const;
+    Q_INVOKABLE QString     parseOptions(QString options);
```
```suggestion
	Q_INVOKABLE QString		parseOptions(QString options);
```

JorisGoosen commented on QMLComponents/analysisform.h:
```diff
@@ -180,9 +181,8 @@ public slots:
-	bool					parseOptions(Json::Value& options);
-	void					setActiveJASPControl(JASPControl* control, bool hasActiveFocus);
-	JASPControl*			getActiveJASPControl()	{ return _activeJASPControl; }
+    void					setActiveJASPControl(JASPControl* control, bool hasActiveFocus);
+    JASPControl*			getActiveJASPControl()	{ return _activeJASPControl; }
```
```suggestion
	void					setActiveJASPControl(JASPControl* control, bool hasActiveFocus);
	JASPControl*			getActiveJASPControl()	{ return _activeJASPControl; }
```

JorisGoosen committed:
Tabs
